### PR TITLE
Support adding static label to document layer

### PR DESCRIPTION
**Pull request by junqiu-lei**
### Description
This PR will allow user to add static label to document layer symbol. User can enable the label, customize the label text tittle, color, size, border color and border width. Its Opacity, zoom range and visibility will be followed with the document layer.

https://user-images.githubusercontent.com/90288540/222991167-9d121de6-67b5-4560-ad5f-e39c12bc67f2.mov

 
### Issues Resolved
https://github.com/opensearch-project/dashboards-maps/issues/321

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/model/mapLayerType.ts [hunk 1]
```diff
@@ -52,6 +52,13 @@ export type DocumentLayerSpecification = {
     borderColor: string;
     borderThickness: number;
     markerSize: number;
+    enableLabel: boolean;
+    labelTittle: string;
+    labelTittleType: 'fixed' | 'by_field';
+    labelColor: string;
+    labelSize: number;
+    labelBorderColor: string;
+    labelBorderWidth: number;
```
Comment (VijayanB) [inline public/model/mapLayerType.ts:61]: What do you think about moving label as nested property below?:

```
label: {
 enableLabel: boolean;
    labelTittle: string;
    labelTittleType: 'fixed' | 'by_field';
    labelColor: string;
    labelSize: number;
    labelBorderColor: string;
    labelBorderWidth: number;
}
```

File: public/components/layer_config/documents_config/style/color_picker.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ */
+
+import React, { memo, useEffect } from 'react';
+import { EuiColorPicker, EuiFormRow, useColorPickerState } from '@elastic/eui';
+import { DocumentLayerSpecification } from '../../../../model/mapLayerType';
+
+export interface ColorPickerProps {
+  originColor: string;
+  label: string;
+  selectedLayerConfig: DocumentLayerSpecification;
```
Comment (VijayanB) [inline public/components/layer_config/documents_config/style/color_picker.tsx:13]: shall we pass only id here?

File: public/model/mapLayerType.ts (hunk 1 above)
Comment (junqiu-lei) [reply public/model/mapLayerType.ts:61]: Ack, will update

File: public/components/layer_config/documents_config/style/color_picker.tsx (hunk 2 above)
Comment (junqiu-lei) [reply public/components/layer_config/documents_config/style/color_picker.tsx]: updated

**Review by VijayanB — APPROVED** [review]

**Review by heemin32 — APPROVED** [review]

**Pull request merged**